org.testeditor.web.dropwizard.xtext/src/main/java/org/testeditor/web/xtext/index/CustomWebResourceSetProvider.xtend

```diff
@@ -0,0 +1,27 @@
+package org.testeditor.web.xtext.index
+
+import javax.inject.Singleton
+import org.eclipse.xtext.resource.XtextResourceSet
+import org.eclipse.xtext.web.server.IServiceContext
+import org.eclipse.xtext.web.server.model.IWebResourceSetProvider
+
+@Singleton
+class CustomWebResourceSetProvider implements IWebResourceSetProvider {
+
+	val XtextIndex index
+
+	new(XtextIndex newIndex) {
```

Comment:
Constructor injection…?

Reply:
Injection turns out not to work here, since this instance is created during guice module setup (maybe I should add this as comment)